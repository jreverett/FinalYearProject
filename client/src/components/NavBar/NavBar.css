#nav-container {
  display: flex;
  height: 100px;
  align-items: center;
  margin-left: 90px;
}

#nav-login-button-container {
  width: 100%;
  text-align: left;
  margin-top: 0;
}

#nav-login-button {
  color: black;
  font-size: large;
}

#img-container {
  display: flex;
  align-items: center;
}
