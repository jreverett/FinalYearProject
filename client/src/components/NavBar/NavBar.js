import React, { Component, Fragment } from 'react';
import { userService } from '../../services/user';
import { NavMenu } from '../index';
import Button from 'react-bootstrap/Button';
import './NavBar.css';

class NavBar extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    var loggedIn = this.props.isLoggedIn;
    return (
      <Fragment>
        <NavMenu />
        <div id="nav-container">
          <div id="nav-login-button-container">
            <Button
              id="nav-login-button"
              href="/login"
              variant="link"
              onClick={() => {
<<<<<<< HEAD
                if (loggedIn) {
                } // userService.logout(email); // TODO: get email prop and link this up
=======
                if (loggedIn) userService.logout;
>>>>>>> 5ad32917a5ac7c11d824f9069eae878505bd06a8
              }}
            >
              {loggedIn && 'LOGOUT'}
              {!loggedIn && 'LOGIN'}
            </Button>
          </div>
          <div id="img-container">
            <a href="/">
              <img src={require('../../images/upvent.png')} />
            </a>
          </div>
        </div>
      </Fragment>
    );
  }
}

export default NavBar;
