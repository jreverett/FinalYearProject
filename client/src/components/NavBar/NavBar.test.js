import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';

describe('<NavBar />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
