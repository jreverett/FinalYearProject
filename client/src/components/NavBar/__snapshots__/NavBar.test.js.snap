// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> matches the snapshot 1`] = `
Array [
  <div>
    <div
      className="bm-overlay"
      onClick={[Function]}
      style={
        Object {
          "MozTransform": "translate3d(100%, 0, 0)",
          "MsTransform": "translate3d(100%, 0, 0)",
          "OTransform": "translate3d(100%, 0, 0)",
          "WebkitTransform": "translate3d(100%, 0, 0)",
          "background": "rgba(0, 0, 0, 0.3)",
          "height": "100%",
          "opacity": 0,
          "position": "fixed",
          "transform": "translate3d(100%, 0, 0)",
          "transition": "opacity 0.3s, transform 0s 0.3s",
          "width": "100%",
          "zIndex": 1000,
        }
      }
    />
    <div
      className="bm-menu-wrap"
      id=""
      style={
        Object {
          "MozTransform": "translate3d(-100%, 0, 0)",
          "MsTransform": "translate3d(-100%, 0, 0)",
          "OTransform": "translate3d(-100%, 0, 0)",
          "WebkitTransform": "translate3d(-100%, 0, 0)",
          "height": "100%",
          "position": "fixed",
          "right": "inherit",
          "transform": "translate3d(-100%, 0, 0)",
          "transition": "all 0.5s",
          "width": "300px",
          "zIndex": 1100,
        }
      }
    >
      <div
        className="bm-menu"
        style={
          Object {
            "boxSizing": "border-box",
            "height": "100%",
            "overflow": "auto",
          }
        }
      >
        <nav
          className="bm-item-list"
          style={
            Object {
              "height": "100%",
            }
          }
        >
          <a
            className="bm-item menu-item"
            href="/"
            id="home"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Home
          </a>
          <a
            className="bm-item menu-item"
            href="/login"
            id="login"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Login
          </a>
          <a
            className="bm-item menu-item"
            href="/event/create"
            id="create-event"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Create Event
          </a>
          <a
            className="bm-item menu-item"
            href="/event-listings"
            id="event-listings"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Events
          </a>
          <a
            className="bm-item menu-item"
            href="/user"
            id="user"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex={-1}
          >
            Profile
          </a>
        </nav>
      </div>
      <div
        style={Object {}}
      >
        <div
          className="bm-cross-button"
          style={
            Object {
              "height": 24,
              "position": "absolute",
              "right": 8,
              "top": 8,
              "width": 24,
            }
          }
        >
          <span
            style={
              Object {
                "position": "absolute",
                "right": "14px",
                "top": "6px",
              }
            }
          >
            <span
              className="bm-cross"
              style={
                Object {
                  "height": 14,
                  "position": "absolute",
                  "transform": "rotate(45deg)",
                  "width": 3,
                }
              }
            />
            <span
              className="bm-cross"
              style={
                Object {
                  "height": 14,
                  "position": "absolute",
                  "transform": "rotate(-45deg)",
                  "width": 3,
                }
              }
            />
          </span>
          <button
            onClick={[Function]}
            style={
              Object {
                "background": "transparent",
                "border": "none",
                "cursor": "pointer",
                "fontSize": 0,
                "height": "100%",
                "left": 0,
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
            tabIndex={-1}
          >
            Close Menu
          </button>
        </div>
      </div>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="bm-burger-button"
        style={
          Object {
            "zIndex": 1000,
          }
        }
      >
        <span>
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "0%",
              }
            }
          />
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "40%",
              }
            }
          />
          <span
            className="bm-burger-bars"
            style={
              Object {
                "height": "20%",
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": "80%",
              }
            }
          />
        </span>
        <button
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "cursor": "pointer",
              "fontSize": 0,
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          Open Menu
        </button>
      </div>
    </div>
  </div>,
  <div
    id="nav-container"
  >
    <div
      id="nav-login-button-container"
    >
      <a
        className="btn btn-link"
        href="/login"
        id="nav-login-button"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        LOGIN
      </a>
    </div>
    <div
      id="img-container"
    >
      <a
        href="/"
      >
        <img
          alt="Upvent logo"
          src="upvent.png"
        />
      </a>
    </div>
  </div>,
]
`;
